sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1260,6 +1263,60 @@ bool _pi_queue::isBatchingAllowed() {
   return (this->QueueBatchSize > 0 && ((ZeSerialize & ZeSerializeBlock) == 0));
 }
 
+pi_result _pi_queue::getOrCreateCopyCommandQueue(
+    int Index, ze_command_queue_handle_t &ZeCopyCommandQueue) {
+  ZeCopyCommandQueue = nullptr;
+
+  // Make sure 'Index' is within limits
+  PI_ASSERT((Index >= 0) && (Index < (int)(ZeCopyCommandQueues.size())),
+            PI_INVALID_VALUE);
+
+  // Return the Ze copy command queue, if already available
+  if (ZeCopyCommandQueues[Index]) {
+    ZeCopyCommandQueue = ZeCopyCommandQueues[Index];
+    return PI_SUCCESS;
+  }
+
+  ZeStruct<ze_command_queue_desc_t> ZeCommandQueueDesc;
+
+  // Ze copy command queue is not available at 'Index'. So we create it below.
+  
+  if (Index == 0) {
+    // Create queue to main copy engine
+    zePrint("NOTE: Main Copy Engine ZeCommandQueueDesc.ordinal = %d, "
+            "ZeCommandQueueDesc.index = %d\n",
+            Device->ZeMainCopyQueueGroupIndex, 0);
+    ZeCommandQueueDesc.ordinal = Device->ZeMainCopyQueueGroupIndex;
+    ZeCommandQueueDesc.index = 0;
+    ZE_CALL(zeCommandQueueCreate,
+            (Context->ZeContext, Device->ZeDevice,
+             &ZeCommandQueueDesc, // TODO: translate properties
+             &ZeCopyCommandQueue));
+    // Main Copy Command Queue is pushed at start of ZeCopyCommandQueues
+    // vector.
+    ZeCopyCommandQueues[0] = ZeCopyCommandQueue;
+    return PI_SUCCESS;
+  } else {
+    // Create a queue to one of link copy engines and copy them into
+    // ZeCopyCommandQueues vector.
+
+    // Index within Link Copy Engines
+    int LinkIndex = Index - Device->hasMainCopyEngine();
+    zePrint("NOTE: Link Copy Engine ZeCommandQueueDesc.ordinal = %d, "
+            "ZeCommandQueueDesc.index = %d\n",
+            Device->ZeLinkCopyQueueGroupIndex, LinkIndex);
+    ZeCommandQueueDesc.ordinal = Device->ZeLinkCopyQueueGroupIndex;
+    ZeCommandQueueDesc.index = LinkIndex;
+    ZE_CALL(zeCommandQueueCreate,
+            (Context->ZeContext, Device->ZeDevice,
+             &ZeCommandQueueDesc, // TODO: translate properties
+             &ZeCopyCommandQueue));
+    ZeCopyCommandQueues[Index] = ZeCopyCommandQueue;
+    return PI_SUCCESS;
+  }
```

Comment:
the if/else above exepctedly do very similar things, can you combine them please?

Reply:
Done. Thanks